File: LanguAI.Backend/LanguAI.Backend/Core/Models/Post.cs
﻿using LanguAI.Backend.Core.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LanguAI.Backend.Core.Models;

public class Post
{
    [Required]
    public int Id { get; set; }

    [Required]
    public string Content { get; set; }

    [Required]
    public int UserId { get; set; }

    [ForeignKey(nameof(UserId))]
    public virtual User User { get; set; }

    [Required]
    public DateTime Created { get; set; }

    [Required]
    public AccessEnum Access { get; set; }

    public virtual ICollection<Interaction> Interactions { get; set; } = new List<Interaction>();
}

File: LanguAI.Frontend/src/api/models/post.ts
/* tslint:disable */
/* eslint-disable */
import { AccessEnum } from '../models/access-enum';
import { Image } from '../models/image';
import { Interaction } from '../models/interaction';
import { User } from '../models/user';
export interface Post {
  access: AccessEnum;
  content: string;
  created: string;
  id: number;
  image?: Image;
  imageId?: number | null;
  interactions?: Array<Interaction> | null;
  isDeleted?: boolean;
  user?: User;
  userId: number;
}
